modelling/src/main/java/org/axonframework/modelling/command/inspection/AggregateModel.java

```diff
@@ -53,12 +54,32 @@
     }
 
     /**
-     * Gets the declared type based on given {@code type}.
+     * Gets the declared aggregate type based on given class {@code type}.
      *
      * @param type the type of the aggregate
      * @return the declared aggregate type represented by {@link String}, if exists
      */
     default Optional<String> declaredType(Class<?> type) {
         return Optional.empty();
     }
+
+    /**
+     * Gets a default constructor for a given class {@code type}.
+     *
+     * @param type the type of the aggregate
+     * @return the default constructor of this {@code type}, if exists
+     */
+    default Optional<Constructor<?>> defaultConstructor(Class<?> type) {
+        return Optional.empty();
+    }
+
+    /**
+     * Checks whether all types in this aggregate have a default constructor (aggregate types defined by an interface
+     * are excluded from this check).
+     *
+     * @return whether all types in this aggregate have a default constructor
+     */
+    default boolean allTypesHaveDefaultConstructor() {
```

Comment:
Same as above. If an AggregateFactory assumes default constructor, the Model should expose all the types declared in this Model, and the AggregateFactory should check for the presence of default constructors, where relevant.